de.dlr.sc.virsat.model.extension.fdir/src/de/dlr/sc/virsat/model/extension/fdir/synthesizer/ASynthesizer.java

```diff
@@ -62,28 +63,32 @@ public RecoveryAutomaton synthesize(Fault fault, Map<ReliabilityRequirement, Fau
 		RecoveryAutomaton synthesizedRA = new RecoveryAutomaton(fault.getConcept());
 		if (modularizer != null) {
 			Set<Module> modules = modularizer.getModules(fault.getFaultTree());
-			Set<Module> trimmedModules = trimStaticModules(modules);
-			
-			Set<RecoveryAutomaton> ras = trimmedModules.stream()
-						.map(module -> convertToRecoveryAutomaton(module))
-						.collect(Collectors.toSet());
+			Set<Module> trimmedModules = ftTrimmer.trimModules(modules);
+			trimmedModules.stream().forEach(module -> module.constructFaultTreeCopy());
 			
-			if (minimizer != null) {
-				ras.forEach(ra -> minimizer.minimize(ra));
+			Set<RecoveryAutomaton> ras = new HashSet<>();
+			for (Module module : trimmedModules) {
+				RecoveryAutomaton ra = convertToRecoveryAutomaton(module);
+				if (minimizer != null) {
+					minimizer.minimize(ra);
+				}
+				
+				Map<FaultTreeNode, FaultTreeNode> mapGeneratedToGenerator = this.mapCopyToOriginalNodes(conversionResult.getMapGeneratedToGenerator(), module.getMapOriginalToCopy());

```

Comment:
I think the `mapCopyToOriginalNodes` method needs a better name.
At least is should contain a verb like create.